frontend/src/metabase/admin/settings/components/ApiKeys/ManageApiKeys.tsx

```diff
@@ -0,0 +1,149 @@
+  return (
+    <>
+      {modal === "create" ? (
+        <CreateApiKeyModal onClose={handleClose} refreshList={refreshList} />
+      ) : modal === "edit" && activeApiKey ? (
+        <EditApiKeyModal
+          onClose={handleClose}
+          refreshList={refreshList}
+          apiKey={activeApiKey}
+        />
+      ) : modal === "delete" && activeApiKey ? (
+        <DeleteApiKeyModal
+          apiKey={activeApiKey}
+          onClose={handleClose}
+          refreshList={refreshList}
+        />
+      ) : null}
+      <Stack pl="md" spacing="lg">
+        <Breadcrumbs
+          crumbs={[
+            [t`Authentication`, "/admin/settings/authentication"],
+            [t`API Keys`],
+          ]}
+        />
+        <Group align="end" position="apart">
+          <Stack>
+            <Title>{t`Manage API Keys`}</Title>
+            {!isShowingEmptyTable && (
+              <Text color="text.1">{t`Allow users to use the API keys to authenticate their API calls.`}</Text>
+            )}
+          </Stack>
+          <Button
+            variant="filled"
+            onClick={() => setModal("create")}
+          >{t`Create API Key`}</Button>
+        </Group>
+        <LoadingAndErrorWrapper loading={isLoading} error={error}>
+          <table
+            className="ContentTable border-bottom"
+            data-testid="api-keys-table"
+          >
+            <thead>
+              <tr>
+                <th>{t`Key name`}</th>
+                <th>{t`Group`}</th>
+                <th>{t`Key`}</th>
+                <th>{t`Last Modified By`}</th>
+                <th>{t`Last Modified On`}</th>
+                <th />
+              </tr>
+            </thead>
+            <tbody>
+              {apiKeys.map(apiKey => (
+                <tr key={apiKey.id} className="border-bottom">
+                  <td className="text-bold">{apiKey.name}</td>
+                  <td>{apiKey.group_name}</td>
+                  <td className="text-monospace">{apiKey.masked_key}</td>
+
+                  {/* TODO: replace with creator_name */}
+                  <td>{apiKey.creator_id}</td>
+
+                  <td>{apiKey.updated_at}</td>
+                  <td>
+                    <Group spacing="md">
+                      <Icon
+                        name="pencil"
+                        className="cursor-pointer"
+                        onClick={() => {
+                          setActiveApiKey(apiKey);
+                          setModal("edit");
+                        }}
+                      />
+                      <Icon
+                        name="trash"
+                        className="cursor-pointer"
+                        onClick={() => {
+                          setActiveApiKey(apiKey);
+                          setModal("delete");
+                        }}
+                      />
+                    </Group>
+                  </td>
+                </tr>
+              ))}
+            </tbody>
+          </table>
```

Comment:
can this table be its own view component? would probably make it easier to unit test

Reply:
09418becd34e8c2e6e332286a9ac8646b5b77ab8

Reply:
1ab2574247898627b92b896316cbfffaf2cb42d8